CDP4WebServices.API.Tests/Services/Authorization/PermissionServiceTestFixture.cs

```diff
@@ -0,0 +1,176 @@
+        private Mock<ParticipantDao> participantDao;
+
+        private AuthenticationPerson authenticationPerson;
+        private Requirement requirement;
+        private Definition definition;
+        private DomainOfExpertise domain;
+        private Participant participant;
+
+        [SetUp]
+        public void TestSetup()
+        {
+            this.authenticationPerson = new AuthenticationPerson(Guid.NewGuid(), 0)
+            {
+                UserName = "TestRunner"
+            };
+
+            this.permissionService = new PermissionService();
+
+            this.permissionService.Credentials = new Credentials
+            {
+                Person = this.authenticationPerson,
+                EngineeringModelSetup = new EngineeringModelSetup(Guid.NewGuid(), 0)
+            };
+
+            this.resolveService = new Mock<IResolveService>();
+
+            this.resolveService.Setup(x => x.ResolveItems(null, It.IsAny<string>(), It.IsAny<Dictionary<DtoInfo, DtoResolveHelper>>()))
+                .Callback<NpgsqlTransaction, string, Dictionary<DtoInfo, DtoResolveHelper>>
+                ((npgsqlTransaction, partition, operationThingContainerCache) =>
+                {
+                    operationThingContainerCache.Add(new ContainerInfo(nameof(Requirement), this.requirement.Iid), new DtoResolveHelper(this.requirement));
+                });
+
+            this.permissionService.ResolveService = this.resolveService.Object;
+
+            this.accessRightKindService = new Mock<IAccessRightKindService>();
+
+            this.permissionService.AccessRightKindService = this.accessRightKindService.Object;
+
+            this.accessRightKindService.Setup(
+                    x =>
+                        x.QueryParticipantAccessRightKind(It.IsAny<Credentials>(), "Definition"))
+                .Returns(ParticipantAccessRightKind.SAME_AS_CONTAINER);
+
+            this.domain = new DomainOfExpertise(Guid.NewGuid(), 0);
+
+            this.participant = new Participant(Guid.NewGuid(), 0)
+            {
+                Domain = new List<Guid> { this.domain.Iid },
+                Person = this.authenticationPerson.Iid
+            };
+
+            this.permissionService.Credentials.EngineeringModelSetup.Participant.Add(this.participant.Iid);
+
+            this.participantDao = new Mock<ParticipantDao>();
+
+            this.participantDao.Setup(
+                    x =>
+                        x.Read(null, It.IsAny<string>(), null, false))
+                .Returns(new List<Participant>() { this.participant });
+
+            this.permissionService.ParticipantDao = this.participantDao.Object;
+
+            this.requirement = new Requirement(Guid.NewGuid(), 0);
+            this.definition = new Definition(Guid.NewGuid(), 0);
+
+            this.requirement.Definition.Add(this.requirement.Iid);
+        }
+
+        [Test]
+        public void VerifySameAsContainerPermissionAutorization()
```

Comment:
can you also add tests for:
  - a Thing that is contained by SiteDirectory
  - a Thing that is contained by EngineeringModel
  - a Thing that is contained by Iteration